pgzx-update-0.2.0.mdx

```diff
@@ -0,0 +1,411 @@
+support is still very rudimentary and you still have to write C-style functions in
+Zig. We plan to provide a more Zig idiomatic API in the future.
+
+When writing an FDW you export 2 functions, the handler function that sets up a
+list of callbacks for the FDW framework and a validator function that verifies
+your FDWs options when users configure the server, user mapping, or foreign
+table. The `pgzx.fdw` modules provides some helpers to declaratively define the
+options your foreign data wrapper supports.
+
+The options API also allows you to combine multiple configurations in case you
+read a list of supported options from a library that you use for your
+implementation.
+
+This example uses reads the available options from libpq into a global table:
+
+```zig
+
+const pgzx = @import("pgzx");
+const fdw = pgzx.fdw;
+
+// postgres_fdw options not related to connection or transaction handling.
+const postgres_fdw_options = fdw.OptionList.init(&[_]fdw.Option{
+    fdw.Option.String("schema_name", pg.ForeignTableRelationId),
+    fdw.Option.String("table_name", pg.ForeignTableRelationId),
+    fdw.Option.String("column_name", pg.AttributeRelationId),
+
+    ...
+});
+
+// internal options for connection and transaction handling.
+const xact_conn_options = fdw.OptionList.init(&[_]fdw.Option{
+    fdw.Option.Bool("keep_connections", pg.ForeignServerRelationId),
+    fdw.Option.Bool("password_required", pg.UserMappingRelationId),
+});
+
+
+// libpq connection options.
+var extension_libpq_options: ?[]fdw.Option = undefined;
+
+
+fn pgdc_fdw_validator(list_datum: pg.Datum, catalog: pg.Oid) void {
+    initPqOptions(); // read libpq options into extension_libpq_options
+
+    const extension_options = fdw.MultiOptionList.init(&[_]fdw.OptionList{
+        postgres_fdw_options,
+        xact_conn_options,
+        fdw.OptionList.init(extension_libpq_options.?),
+    });
+
+    const options = pg.untransformRelOptions(list_datum);
+    fdw.validateOptions(extension_options, options, catalog);
+}
+```
+
+The handler function can be implemented as follows:
+
+```zig
+fn pgdc_fdw_handler() pg.Datum {
+    return pg.PointerGetDatum(pgzx.node.create(pg.FdwRoutines{
+          .BeginForeignScan = ...,
+          .IterateForeignScan = ...,
+          .EndForeignScan = ...,
+          ...
+    }));
+}
+```
+
+## Other changes
+
+- Add some more utilities to pgzx.meta module.
+- We updated the `pgzx.datum` module to also accept the type OID for a value as
+  hint to the converter. This is especially important when working with
+  strings, as different strings types might be encoded differently. We also
+  export a number of helper functions for converting between pg.Datum and Zig
+  strings in case the derived converter does not work well for you.
+- Collections: Added `at` to `List` and `valuesIterator` to `HTab`.
+- add `pgzx.node` module for working with the PostgreSQL `Node` type.
+- Build: we split the C header includes into a separate module named `pgsys`.
```

Comment:
```suggestion
- Build: we split the C header included into a separate module named `pgsys`.
```

Reply:
why included? When working with C one uses `include` to reference a file (the contents of the other file is copied in place). I mean `includes` as a multiple included objects.

Reply:
I'm not sure I understand. It just reads odd to me. What about "we split the C header _to include_ a separate module..." ?

Reply:
that wouldn't be what we did. How about 'We moved the C header includes into its own Zig module `pgsys`'?